modules/context/repo.go

```diff
@@ -440,6 +440,25 @@ func RepoAssignment(ctx *Context) (cancel context.CancelFunc) {
 	ctx.Repo.Owner = owner
 	ctx.Data["Username"] = ctx.Repo.Owner.Name
 
+	// redirect link to wiki
+	if strings.HasSuffix(repoName, ".wiki") {
+		parts := strings.Split(ctx.Req.RequestURI, "/")
+		for i := len(parts) - 1; i >= 0; i-- {
+			if parts[i] == repoName {
+				parts[i] = strings.Replace(parts[i], ".wiki", "/wiki", 1)
+				break
+			}
+		}
+		
+		redirectLink := ""
+		for _, part := range parts {
+			redirectLink += part + "/"
+		}
```

Comment:
https://pkg.go.dev/net/url#URL.RequestURI

Bugs:

case 1: `/path?foo=bar` => `/path?foo=bar/`
case 2: `/path/` => `/path//`

And the code above should not split RequestURI directly IMO.